lib/trino-parquet/src/main/java/io/trino/parquet/predicate/TupleDomainParquetPredicate.java

```diff
@@ -392,10 +392,13 @@ private static Domain getDomain(
             }
             else {
                 for (int i = 0; i < minimums.size(); i++) {
-                    Int128 min = Int128.fromBigEndian(((Slice) minimums.get(i)).getBytes());
-                    Int128 max = Int128.fromBigEndian(((Slice) maximums.get(i)).getBytes());
+                    Object min = minimums.get(i);
+                    Object max = maximums.get(i);
 
-                    rangesBuilder.addRangeInclusive(min, max);
+                    Int128 minValue = min instanceof Slice ? Int128.fromBigEndian(((Slice) min).getBytes()) : Int128.valueOf(asLong(min));
```

Comment:
We could use enhanced switch expression here

Reply:
We could, but feels unnecessary for one condition